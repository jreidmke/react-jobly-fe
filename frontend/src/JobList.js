import Job from './Job';
import { useEffect, useState, useContext } from 'react';
import {useHistory} from 'react-router-dom';import JoblyApi from './api';
import Search from './Search';
import UserContext from './UserContext';

const JobList = () => {
    const history = useHistory();
    const {currUser} = useContext(UserContext);
    
    if(!currUser) {
        history.push('/');
    }

    const [jobs, setJobs] = useState([]);

    useEffect(() => {
        async function getJobs() {
            const jobs = await JoblyApi.getAllJobs();
            setJobs(jobs);
        }
        getJobs();
    }, []);

    async function search(formData) {
        const searchRes = await JoblyApi.getJobs(formData);
        setJobs(searchRes);
    }

    return(
        <div>
            <Search handleSearch={search}/>
            {jobs ? jobs.map(j => (
                <Job company={j.companyName} title={j.title} salary={j.salary} equity={j.equity} id={j.id}/>
            )) : "Loading..."}
        </div>
    )
}

export default JobList;