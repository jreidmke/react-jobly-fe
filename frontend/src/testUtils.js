import React from "react";
import UserContext from "./UserContext";

const demoUser = {username: "jreid", firstName: "James", lastName: "Reid", email: "jreidmke@gmail.com", isAdmin: false, applcations: []};

const UserProvider =
    ({ children, currUser = demoUser, hasAppliedToJob = () => false }) => (
    <UserContext.Provider value={{ currUser, hasAppliedToJob }}>
      {children}
    </UserContext.Provider>
);

export { UserProvider };
