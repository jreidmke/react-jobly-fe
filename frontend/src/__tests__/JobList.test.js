import React from "react";
import { render, waitFor, fireEvent } from "@testing-library/react";
import JobList from "../JobList";
import { MemoryRouter } from "react-router";
import { UserProvider } from "../testUtils";

it("renders without crashing", () => {
    render(
        <UserProvider>
            <MemoryRouter>
                <JobList/>
            </MemoryRouter>
        </UserProvider>          
    )
});

it("matches the snapshot", () => {
    const {asFragment} = render(
        <UserProvider>
            <MemoryRouter>
                <JobList/>
            </MemoryRouter>
        </UserProvider>          
    );
    expect(asFragment()).toMatchSnapshot();
}); 

it("returns correct jobs when searched", async() => {
    const {getByPlaceholderText, getByText, queryByText} = render(
        <UserProvider>
            <MemoryRouter>
                <JobList/>
            </MemoryRouter>
        </UserProvider>          
    );
    const button = getByText("Search");
    const input = getByPlaceholderText("Search Here");
    fireEvent.change(input, { target: { value: 'on' }});
    fireEvent.click(button);
    // const conservator = waitFor(() => queryByText("Conservator"));
    // waitFor(() => expect(conservator).toBeInTheDocument());
})