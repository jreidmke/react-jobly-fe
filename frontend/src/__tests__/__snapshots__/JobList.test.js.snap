// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<DocumentFragment>
  <div>
    <form>
      <input
        id="search-term"
        name="searchTerm"
        placeholder="Search Here"
        type="text"
      />
      <button>
        Search
      </button>
    </form>
  </div>
</DocumentFragment>
`;
